### ISSUE TITLE
Chroma key video texture sample not working correctly

### ISSUE THREAD

**Issue by Venthorus** [ThomasGorisse/sceneform-android-sdk#79]
Hello,

the chroma key video texture sample seems to be broken since v1.18.10; it currently looks more like a mirror of your surroundings:

![video_texture_chroma_key](https://user-images.githubusercontent.com/7419985/121335994-63c3ac00-c91b-11eb-9436-8734fe4b0373.png)

It worked correctly in v1.18.9. The plain video texture sample works fine in all versions however.

I'm not a 3D expert, but I suppose it may have something to do with the material and/or the material fixes that went into v1.18.10.

**Comment by ThomasGorisse** [comment]
Hi,
Thanks for this report. You are right.
Blame it on me, I made one mistake during the material fixes commits.
We are currently merging a great improvement from @RGregat and I than add the fix to this issue.

**Comment by Venthorus** [comment]
No worries, thanks for the fix. I really appreciate the effort.

I actually saw the commented line in setMaterial when I was browsing RenderableInstance while trying to understand how things connect with Filament and I was wondering if that looked right. Funny that indeed it was part of the problem.

**Comment by ThomasGorisse** [comment]
That was it.
If you want to contribute, it will be awesome to have a VideoNode which includes all the boilerplate code of the video-texture sample. Specially the material part which should be common to most video textures.
Interested ?

**Comment by Venthorus** [comment]
I would gladly help with that. I did something similar with the original Sceneform and this time too so I should be able to contribute.
I already have several ideas on how the API may look like. In the easiest case someone just adds a VideoNode with a raw resource ID and calls it a day.

I will get back to you in the next few days.

**Comment by ThomasGorisse** [comment]
Cool. I gave you contributor access. You can make the PR directly on a branch of this repo. It's quite easy from GitHub or Android Studio directly.

For the VideoNode, I really think that it will be more accurate to pass the MediaPlayer to the constructor instead of a res id since it let the ability for the user to manage the playing state and other things.

Thanks

**Comment by ThomasGorisse** [comment]
After asking to the Filament team [here](https://github.com/google/filament/discussions/4118), we should use matc compiled .filamat for the external video material instead of writing the shader directly in the code.
If you need help for this, even if I am not an expert on coding shaders, I can generate the plain and chroma key .filamat ones.
You'll just have to call the `Material.builder().setSource(Context context, Uri sourceUri)` in place of the actual material configuration.

**Comment by Venthorus** [comment]
I tried it out myself and I was able to run the matc tool (from Filament 1.9.25, because there actually is a material breakage in 1.10.0) and compile the chroma key mat file (the one that is in the repository) to a filamat file (for mobile). When building the material with the file it doesn't throw an error, however I don't see the video. The model is there as I can't click behind it and the video is running because I can hear it.

I'm also not into shaders, but in this case I don't understand why it is not showing up. The parameters match up (and yes, in the file it is called "keyColor" instead of "chromaKeyColor") and the code part is the same as the one that was generated directly in code. If I switch to the material written in code it works again. Am I missing something?
